File: src/ransom-note/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        hashmap = {}

        for ch in magazine:
            hashmap[ch] = hashmap.get(ch, 0) + 1

        for ch in ransomNote:
            if hashmap.get(ch, 0) <= 0:
                return False
            hashmap[ch] = hashmap.get(ch, 0) - 1

        return True

File: src/ransom-note/ransom-note.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    const hashmap: Record<string, number> = {};
    for (const ch of magazine) {
        hashmap[ch] = (hashmap[ch] || 0) + 1;
    }
    for (const ch of ransomNote) {
        if (!hashmap[ch]) {
            return false;
        }
        hashmap[ch]--;
    }
    return true;
}